### PR TITLE
AWS Lamda SDK: Upgrade input/output body stripping

### PR DESCRIPTION
**Pull request by medikoo**
- Ensure not to resolve request/response payload when it's not meant to send
- Strip binary lambda request
- Strip binary lambda large requests and responses
- Strip AWS SDK large requests and responses
- Notify with notice events body stripping

### CODE REVIEW

File: node/packages/aws-lambda-sdk/instrument/index.js [hunk 1]
```diff
@@ -28,6 +28,31 @@ const { awsLambda: awsLambdaSpan, awsLambdaInitialization: awsLambdaInitializati
...
+        serverlessSdk._reportNotice('Binary body excluded', `${prefix}_BODY_BINARY`, {
+          _traceSpan: awsLambdaSpan,
+        });
+      }
+    }
+  }
+  if (
+    data.body &&
+    Buffer.byteLength(JSON.stringify(data.body)) > serverlessSdk._maximumBodyByteLength
+  ) {
```
Comment (Danwakeem) [inline node/packages/aws-lambda-sdk/instrument/index.js]: Wouldn't we also want to check if `data` by itself is too large? I am thinking of the scenario when you might be passing in a custom object as an input to your lambda function.

**Review by Danwakeem — COMMENTED** [review]
One question 👍

File: node/packages/aws-lambda-sdk/instrument/index.js (hunk 1 above)
Comment (medikoo) [reply node/packages/aws-lambda-sdk/instrument/index.js]: @Danwakeem that's a very good point. We should actually validate the whole request size, and if it's too large, return null. I've updated PR

File: node/packages/aws-lambda-sdk/instrument/index.js [hunk 2]
```diff
@@ -28,6 +28,29 @@ const { awsLambda: awsLambdaSpan, awsLambdaInitialization: awsLambdaInitializati
 
 const toProtobufEpochTimestamp = awsLambdaSpan.constructor._toProtobufEpochTimestamp;
 
+const resolveBodyString = (data, prefix) => {
+  if (data === undefined) return null;
+  data = { ...data };
```
Comment (Danwakeem) [inline node/packages/aws-lambda-sdk/instrument/index.js]:
```suggestion
```
Do we need to do a spread here?

File: node/packages/aws-lambda-sdk/instrument/index.js (hunk 1 above)
Comment (Danwakeem) [reply node/packages/aws-lambda-sdk/instrument/index.js]: Ahh okay awesome just left one more comment 🙏

File: node/packages/aws-lambda-sdk/instrument/index.js (hunk 2 above)
Comment (medikoo) [reply node/packages/aws-lambda-sdk/instrument/index.js]: Indeed more optimal would be to make it only when we strip the binary body (and in that case is needed, as the same object is processed by user code). Updated

Comment (Danwakeem) [reply node/packages/aws-lambda-sdk/instrument/index.js]: Ahh got it, you wanted to do a copy when you `delete data.body` that way you do not mutate the input param and only effect the output of the function. I think where you moved this spread to makes the most sense 👍

**Review by Danwakeem — APPROVED** [review]
Looking good :shipit:

**Pull request merged**